File: app/src/main/java/com/example/fingertapping/MainActivity.java
package com.example.fingertapping;

import android.Manifest;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.support.v4.app.ActivityCompat;
import android.support.v4.content.ContextCompat;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

import com.google.common.collect.Lists;

import java.util.ArrayList;
import java.util.Collections;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        if (ContextCompat.checkSelfPermission(this,
                Manifest.permission.WRITE_EXTERNAL_STORAGE)
                != PackageManager.PERMISSION_GRANTED) {
            ActivityCompat.requestPermissions(this,
                    new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE},
                    1);

        }

        if (ContextCompat.checkSelfPermission(this,
                Manifest.permission.READ_EXTERNAL_STORAGE)
                != PackageManager.PERMISSION_GRANTED) {

            ActivityCompat.requestPermissions(this,
                    new String[]{Manifest.permission.READ_EXTERNAL_STORAGE},
                    1);
        }

        if (ContextCompat.checkSelfPermission(this,
                Manifest.permission.INTERNET)
                != PackageManager.PERMISSION_GRANTED) {

            ActivityCompat.requestPermissions(this,
                    new String[]{Manifest.permission.INTERNET},
                    1);
        }

    }


    public void tapTwoClicked(View view) {
        ArrayList<Integer> measures = Lists.newArrayList(0, 1, 2);
        Collections.shuffle(measures);
        int[] order = new int[3];
        order[0] = measures.get(0);
        order[1] = measures.get(1);
        order[2] = measures.get(2);
        Intent i = new Intent(getBaseContext(), UserDataActivity.class);
        i.putExtra("order", order);
        startActivity(i);
    }


    public void settingsClicked(View view) {
        Intent i = new Intent(getBaseContext(), SettingsActivity.class);
        startActivity(i);
    }
}

File: app/src/main/java/com/example/fingertapping/MainActivity.kt
package com.example.fingertapping

import android.Manifest
import android.content.Intent
import android.content.pm.PackageManager
import android.os.Bundle
import android.support.v4.app.ActivityCompat
import android.support.v4.content.ContextCompat
import android.support.v7.app.AppCompatActivity
import android.view.View

import com.google.common.collect.Lists

import java.util.ArrayList
import java.util.Collections

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        grantPermission(Manifest.permission.WRITE_EXTERNAL_STORAGE)
        grantPermission(Manifest.permission.READ_EXTERNAL_STORAGE)
        grantPermission(Manifest.permission.INTERNET)
    }

    private fun grantPermission(permission: String) {
        if (ContextCompat.checkSelfPermission(this,
                        permission) != PackageManager.PERMISSION_GRANTED) {
            ActivityCompat.requestPermissions(this,
                    arrayOf(permission),
                    1)

        }
    }


    fun tapTwoClicked(view: View) {
        val measuresRight = generateMeasurementsOrder()
        val measuresLeft = generateMeasurementsOrder()
        val order = createMeasurementsOrderArray(measuresRight, measuresLeft)
        val i = Intent(baseContext, UserDataActivity::class.java)
        i.putExtra("order", order)
        startActivity(i)
    }

    private fun createMeasurementsOrderArray(measuresRight: ArrayList<Int>, measuresLeft: ArrayList<Int>): IntArray {
        val measurementsOrder = IntArray(6)
        measurementsOrder[0] = measuresRight[0]
        measurementsOrder[1] = measuresRight[1]
        measurementsOrder[2] = measuresRight[2]
        measurementsOrder[3] = measuresLeft[0]
        measurementsOrder[4] = measuresLeft[1]
        measurementsOrder[5] = measuresLeft[2]
        return measurementsOrder
    }

    private fun generateMeasurementsOrder(): ArrayList<Int> {
        val measurements = Lists.newArrayList(0, 1, 2)
        measurements.shuffle()
        return measurements
    }


    fun settingsClicked(view: View) {
        val i = Intent(baseContext, SettingsActivity::class.java)
        startActivity(i)
    }
}
